src/i2c/master.rs

```diff
@@ -127,9 +148,19 @@ impl<'a> I2cMaster<'a, Blocking> {
         sda: Peri<'a, impl SdaPin<T>>,
         // TODO - integrate clock APIs to allow dynamic freq selection | clock: crate::flexcomm::Clock,
         speed: Speed,
+        scl_low_clocks: Mstscllow,
+        scl_high_clocks: Mstsclhigh,
     ) -> Result<Self> {
         force_clear_remediation(&T::info());
-        Ok(Self::new_inner::<T>(fc, scl, sda, speed, None)?)
+        Ok(Self::new_inner::<T>(
+            fc,
+            scl,
+            sda,
+            speed,
+            scl_low_clocks,
+            scl_high_clocks,
+            None,
```

Comment:
I think we've delayed it long enough; These arguments should be part of a `struct Config` to make it easier for the user. Look at other HALs in upstream embassy and the eSPI driver for example

Reply:
Sure, moved the 'C-enum' args to a struct. I left the 'which pins to bind to' piece out of the config because I don't think there are sane defaults and they're templated so we'd have to template the config struct, which I think could get messy? Let me know if you think those should move into the struct too and we can discuss